x-pack/filebeat/module/azure/activitylogs/config/azure-eventhub.yml

```diff
@@ -20,6 +19,14 @@ storage_account: {{ .storage_account }}
 storage_account_key: {{ .storage_account_key }}
 {{ end }}
 
+{{ if .storage_account_container }}
+storage_account_container: {{ .storage_account_container }}
+{{ else }}
+{{ if .eventhub }}
+storage_account_container: filebeat-activitylogs-{{ .eventhub }}
```

Comment:
these `else if` will change the behaviour of https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/input/azureeventhub/config.go#L42, or am I wrong?

is it intended?

also regarding to underscore in eventhub we should probably do a string replace always on https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/input/azureeventhub/config.go#L42

Reply:
The log types [activitylogs](https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/module/azure/activitylogs/manifest.yml#L6-L7), [auditlogs](https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/module/azure/auditlogs/manifest.yml#L6-L7), and [signinlogs](https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/module/azure/signinlogs/manifest.yml#L6-L7) have a default value for the eventhub config option. 

For these three cases, the `SAContainer` at [line 42](https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/input/azureeventhub/config.go#L42) should never be empty.

For the [platformlogs](https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/module/azure/platformlogs/manifest.yml#L6), not giving eventhub a value, and with no default, should result in [returning an error](https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/input/azureeventhub/config.go#L36).

Reply:
> also regarding to underscore in eventhub we should probably do a string replace always on

If I explicitly set the eventhub with value that is not valid for an SA container (for example, `test_logs`), there is a check that kicks in to inform the user:

https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/input/azureeventhub/config.go#L44

I was also thinking about a string replace, but these cases can turn themselves into rabbit holes because of the nuances of the checks. For example, according to the [Azure documentation for container names](https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names), they must comply with the following restrictions:

>
> A container name must be a valid DNS name, conforming to the following naming rules:
> 
> 1. Container names must start or end with a letter or number, and can contain only letters, numbers, and the dash (-) character.
>
> 2. Every dash (-) character must be immediately preceded and followed by a letter or number; consecutive dashes are not permitted in container names.
>
> 3. All letters in a container name must be lowercase.
>
>4. Container names must be from 3 through 63 characters long.
>

Replacing `_` with `-` can probably work in most cases, but you can expose yourself to edge cases like `test__logs` where after the replace you end up with a two consecutive dashes, that are also not allowed.

So, mi stance on this topic was to rely on the current strategy of returning an error to the human to pick the right name for the eventhub and the SA container. 

But I'm open to suggestions, let me know what you think. 

Maybe we can improve the error message, including a link to the Azure docs.

Reply:
I want to check under which condition the code at https://github.com/elastic/beats/blob/0192a90f2481602524ef0793f3972ea4368dc042/x-pack/filebeat/input/azureeventhub/config.go#L42

is reachable.

Reply:
I missed this https://github.com/elastic/beats/pull/31279/files#diff-260b477483948aeade35b436549c6e7cb3a2b175569ef4b6c6c82d2c0bcd9f13L4 in the diff, that's why my comment, sorry :)

keeping the current validation is fine for me as well

Reply:
Thank you for taking the time to check the change in detail 💌